### PR TITLE
[GL-4715] Model joins documentation, first pass

### PR DESCRIPTION
**Pull request by wilson-glean**
First pass on basic documentation for model joins.

Adds a `Model Joins` page under `Data Models`, and an addendum to the custom metric SQL section.

Most of the documentation is around setting up the primary and foreign keys, since everything downstream of that should just be the same as existing behavior.

Note - also a bit of data-ops documentation in https://github.com/gleannyc/glean/pull/8861

notes
- "dimensional"? from carlos
- decide if we want to do the canonical example
   - e.g. Customers and Sales

### CODE REVIEW

**Review by vndrewlee — APPROVED** [review]
left a question about the working on alias custom metrics

File: pages/docs/data-modeling/model-joins.mdx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+2. Ensure you have defined an attribute for the primary key to be assigned to.
+
+3. Open the attribute editor and click `Set as primary key`. Uniqueness will also be validated at this step.
+
+<img
+  src="/model_joins/pk.png"
+  alt="Setting a primary key"
+  style={{ padding: "20px" }}
+/>
```
Comment (glean-charlie) [inline pages/docs/data-modeling/model-joins.mdx:39]: this image doesn't actually highlight where the checkbox is. Draw a box over it or something?

**Review by glean-charlie — APPROVED** [review]

File: pages/docs/data-modeling/model-joins.mdx [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import { Callout } from "nextra-theme-docs";
+
+# Model Joins
+
+In many cases, it can be useful to reuse data from one data model in others. This can be done in Glean by configuring joins across models. In essence, this is defining a [primary-foreign key relationship](https://docs.oracle.com/cd/E12100_01/books/admintool/admintool_DataModeling4.html#:~:text=A%20primary%20key%2Dforeign%20key,key%20columns%20in%20another%20table.) across two Glean models.
```
[the remark this answers is not in the document]
Comment (wilson-glean) [reply pages/docs/data-modeling/model-joins.mdx:5]: Just gonna remove the link, I think it's unnecessary

**Comment by glean-charlie** [comment]
🟢 You are good to merge

**Pull request merged**